eox_nelp/programs/api/v1/views.py

```diff
@@ -106,3 +144,56 @@ def post(self, request, course_id):
         update_program_metadata(course_id, serializer.data, request.user)
 
         return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+
+class ProgramsListView(CourseListView):
+    """
+    API view to list courses with program metadata.
+    **Use Cases**
+        - List visible courses with associated program metadata for request user
+        - List courses in which the user identified by national_id is enrolled,
+          provided the requesting user has permission to view them.
+
+    **Example Requests**
+        GET /eox-nelp/api/programs/v1/program-lookup/
+        GET /eox-nelp/api/programs/v1/program-lookup/?national_id=1234567890
+    """
+    authentication_classes = [JwtAuthentication, SessionAuthenticationAllowInactiveUser]
+    permission_classes = [IsAuthenticated, ProgramsLookupPermission]
+    serializer_class = ProgramLookupSerializer
+
+    @process_national_id_query_param()
+    def get(self, request, *args, **kwargs):
+        """Override get to apply some filtering and pre processing."""
+        return super().get(request, *args, **kwargs)
+
+    def get_queryset(self):
+        """Override qs to query by national_id if provided.
+        Returns:
+            QuerySet: queryset of courses depending on national_id presence.
+            If national_id is provided, returns courses the user nid.
+            If not provided, returns all courses visible to the request user.
+        """
+        if getattr(self.request, 'nid_user', None):
+            return courses.get_courses(user=self.request.nid_user)
+        return super().get_queryset()  # Visible courses for user queryset
```

Comment:
do both statements  return the same type of data ?

Reply:
There are two cases that we should clarify
1. `nid_user` has access  to a specific course that `request.user` don't,  should request.user be able to see that course? is `get_courseware_courses() `tenant aware ?
2. if `nid_user` was enrolled in a course but later the visibility of that course changed(Now it's just visible by admins or staff members) should we return that course ?